src/Extension.cpp

```diff
@@ -0,0 +1,337 @@
+      }
+    }
+  }
+
+  auto &variadic = node["variadic"];
+  if (variadic) {
+    auto &min = variadic["min"];
+    auto &max = variadic["max"];
+    if (min) {
+      function.variadic = std::make_optional<FunctionVariadic>(
+          {min.as<int>(),
+           max ? std::make_optional<int>(max.as<int>()) : std::nullopt});
+    } else {
+      function.variadic = std::nullopt;
+    }
+  } else {
+    function.variadic = std::nullopt;
+  }
+
+  return true;
+}
+
+template <> struct convert<EnumArgument> {
+  static bool decode(const Node &node, EnumArgument &argument) {
+    // 'options' is required property
+    auto &options = node["options"];
+    if (options && options.IsSequence()) {
+      auto &required = node["required"];
+      argument.required = required && required.as<bool>();
+      return true;
+    } else {
+      return false;
+    }
+  }
+};
+
+template <> struct convert<ValueArgument> {
+  static bool decode(const Node &node, ValueArgument &argument) {
+    auto &value = node["value"];
+    if (value && value.IsScalar()) {
+      auto valueType = value.as<std::string>();
+      argument.type = Type::decode(valueType);
+      return true;
+    }
+    return false;
+  }
+};
+
+template <> struct convert<TypeArgument> {
+  static bool decode(const Node &node, TypeArgument &argument) {
+    // no properties need to populate for type argument, just return true if
+    // 'type' element exists.
+    return node["type"];
+  }
+};
+
+template <> struct convert<ScalarFunctionVariant> {
+  static bool decode(const Node &node, ScalarFunctionVariant &function) {
+    return decodeFunctionVariant(node, function);
+  };
+};
+
+template <> struct convert<AggregateFunctionVariant> {
+  static bool decode(const Node &node, AggregateFunctionVariant &function) {
+    const auto &res = decodeFunctionVariant(node, function);
+    if (res) {
+      const auto &intermediate = node["intermediate"];
+      if (intermediate) {
+        function.intermediate = Type::decode(intermediate.as<std::string>());
+      }
+    }
+    return res;
+  }
+};
+
+template <> struct convert<io::substrait::TypeVariant> {
+  static bool decode(const Node &node, io::substrait::TypeVariant &typeAnchor) {
+    auto &name = node["name"];
+    if (name && name.IsScalar()) {
+      typeAnchor.name = name.as<std::string>();
+      return true;
+    }
+    return false;
+  }
+};
+
+} // namespace YAML
+
+namespace io::substrait {
+
+namespace {
+
+std::string getSubstraitExtensionAbsolutePath() {
+  const std::string absolute_path = __FILE__;
+  auto const pos = absolute_path.find_last_of('/');
+  return absolute_path.substr(0, pos) + "/../third_party/substrait/extensions/";
+}
+
+} // namespace
+
+std::shared_ptr<Extension> Extension::load() {
+  static const auto registry = loadDefault();
+  return registry;
+}
+
+std::shared_ptr<Extension> Extension::loadDefault() {
+  static const std::vector<std::string> extensionFiles = {
```

Comment:
These yaml files is not migrated into substait-cpp.

Reply:
it's load from substrait submodule